services/dqa-secondary-reports/main.py

```diff
@@ -0,0 +1,271 @@
+    """
+
+    from docopt import docopt
+
+    opts = docopt(usage)
+
+    host = opts['--host']
+    port = int(opts['--port'])
+    debug = opts['--debug']
+    interval = int(opts['--interval'])
+
+    # Set token if defined.
+    if opts['--token']:
+        GITHUB_AUTH_TOKEN = opts['--token']
+    else:
+        GITHUB_AUTH_TOKEN = os.environ.get('GITHUB_AUTH_TOKEN')
+
+    if not GITHUB_AUTH_TOKEN:
+        print('Authorization token required.')
+        sys.exit(1)
+
+    if debug:
+        logger.setLevel(logging.DEBUG)
+
+    dqa_resource = DQAResource()
+
+    # Do the initial fetch of the repository.
+    dqa_resource.update()
+
+    # Create the timer for repo update, and start it.
+    timer = ResourceTimer(dqa_resource, interval)
+    timer.start()
+
+    # Initialize the flask app and register the routes.
+    app = Flask(__name__)
+
+    CORS(app)
+
+    # Unique service ID and timestamp when it started.
+    SERVICE_ID = str(uuid.uuid4())
+    SERVICE_TIMESTAMP = datetime.now().strftime('%Y-%m-%dT%H:%M:%S')
+
+    @app.route('/', methods=['GET'])
+    def index():
+        return json.dumps({
+            'name': 'DQA Service',
+            'version': '0.0.1',  # TODO add version.py
+            'time': SERVICE_TIMESTAMP,
+            'uuid': SERVICE_ID,
+        })
+
+    for version in dqa_resource.get_versions():
+        app.add_url_rule('/pedsnet/' + version + '/field-totals',
+                         version + '_' + 'field-totals',
+                         dqa_resource.get_field_totals(version),
+                         methods=['GET'])
+
+        app.add_url_rule('/pedsnet/' + version + '/site-totals',
+                         version + '_' + 'site-totals',
+                         dqa_resource.get_site_totals(version),
+                         methods=['GET'])
+
+        app.add_url_rule('/pedsnet/' + version + '/table-totals',
+                         version + '_' + 'table-totals',
+                         dqa_resource.get_table_totals(version),
+                         methods=['GET'])
+
+    app.add_url_rule('/dictionary', 'dictionary', dqa_resource.get_dict,
+                     methods=['GET'])
+
+    try:
+        '''
+        Set use_reloader=False when starting the app, otherwise we end up with two resourse update timers
+        starting and then stepping on each other's toes.
+
+        For explanation, see:
+        http://stackoverflow.com/questions/25504149/why-does-running-the-flask-dev-server-run-itself-twice
+
+        The Werkzeug reloader spawns a child process so that it can restart that process each time 
+        your code changes. 
+        Werkzeug is the library that supplies Flask with the development server when you call app.run().
+        See the restart_with_reloader() function code; your script is run again with subprocess.call().
+
+        If you set use_reloader to False you'll see the behaviour go away, but then you also lose the 
+        reloading functionality.
```

Comment:
But I've seen pull requests fail with errors when I had the two threads running at the same time. I don't want to miss some valid errors because of that noise.